modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -265,13 +266,29 @@ public static String toHexString(long addr, int len) {
         StringBuilder sb = new StringBuilder(len * 2);
 
         for (int i = 0; i < len; i++) {
-            // Can not use getLong because on little-endian it produces bs.
+            // Can not use getLong because on little-endian it produces wrong result.
             addByteAsHex(sb, GridUnsafe.getByte(addr + i));
         }
 
         return sb.toString();
     }
 
+    /**
+     * Returns hex representation of memory region.
+     *
+     * @param buf Buffer which content should be converted to string.
+     */
+    public static String toHexString(ByteBuffer buf) {
+        StringBuilder sb = new StringBuilder(buf.capacity() * 2);
+
+        for (int i = 0; i < buf.capacity(); i++) {
```

Comment:
Should't you start with position and end with limit? Please reflect in javadoc that position and limit are ignored and the whole data chunk is converted to string

Reply:
You're right, corrected and expanded the tests.